src/translate.py

```diff
@@ -1673,14 +1691,14 @@ def deref(
             type = new_type
 
     # Dereferencing pointers of known types
-    target = get_pointer_target(var.type, typemap)
-    if field_name is None and target is not None:
-        sub_size, sub_type = target
-        if sub_size == size and offset % size == 0:
+    target = var.type.get_pointer_target()
+    if field_name is None and target is not None and (target.size or target.is_ctype()):
```

Comment:
why is this check needed?

Reply:
You're right, I can remove the `target.is_ctype()` check. 
(The `target.size` is needed because it can be 0, and we try computing `offset % target.size`)

Reply:
I'm only seeing `offset % size`, not `offset % target.size`, and `size` should be statically known to be non-zero at this point. (I also feel like `.size` is a private property of Type, and its semantics are kinda weird, so seems iffy to base logic on)

Reply:
Oh ha, I forgot why I put this in (I should at the very least put in a `TODO` comment!) 

The `target.is_ctype()` is there to port over the old logic that was in `get_pointer_target()`: https://github.com/matt-kempster/mips_to_c/pull/106/files/eace377752e41206c3c85bd26c896f28606278b1#diff-5e877229fb015bfa4a784ad6bca335f1baccf27a146db474ebc04266b358ea19L399

If I remove it, honestly the output gets a bit better... but it's mostly unrelated to what I was trying to change here, and I didn't want to confuse myself. 

Removing the check results in a few diffs like this, in the `break` end-to-end test:
```diff
-        if (D_4101D0.unk4 == 2) {
-            D_4101D0.unk8 = 3;
-        } else if (D_4101D0.unk8 == 2) {
-            D_4101D0.unkC = 3;
+        if ((&D_4101D0)[1] == 2) {
+            (&D_4101D0)[2] = 3;
+        } else if ((&D_4101D0)[2] == 2) {
+            (&D_4101D0)[3] = 3;
```

I'm going to leave it in for now, with a TODO?

Reply:
Oh yeah, you're right, I'll fiddle with this a bit more.

Reply:
Sure!